droidlet/dashboard/web/src/components/Timeline/Timeline.js

```diff
@@ -88,35 +103,42 @@ class DashboardTimeline extends React.Component {
 
   handleClick(item) {
     const eventObj = JSON.parse(item.title);
-    let tableArr = [];
-    for (let key in eventObj) {
-      if (eventObj.hasOwnProperty(key)) {
-        // stringify JSON object for logical form
-        if (key === "logical_form") {
-          tableArr.push({
-            event: this.capitalizeEvent(key),
-            description: JSON.stringify(eventObj[key]),
-          });
-        } else {
-          tableArr.push({
-            event: this.capitalizeEvent(key),
-            description: eventObj[key],
-          });
-        }
-      }
-    }
-    this.setState({
-      tableBody: tableArr,
-    });
+    let tableArr = jsonToArray(eventObj);
+    this.props.stateManager.memory.timelineDetails = tableArr;
+    this.props.stateManager.updateTimeline();
   }
 
-  capitalizeEvent(str) {
-    // replaces underscores with spaces
-    str = str.replace(/_/g, " ");
-    // capitalizes the first letter of every word
-    return str.replace(/\w\S*/g, function (txt) {
-      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
-    });
+  handleSearch(pattern) {
+    if (pattern) {
+      const fuseOptions = {
+        ignoreLocation: true,
+        useExtendedSearch: true,
+      };
+
+      const fuse = new Fuse(
+        this.props.stateManager.memory.timelineEventHistory,
+        fuseOptions
+      );
+
+      const result = fuse.search("'" + pattern);
```

Comment:
curious: why prepend "'" to the pattern?

Reply:
Sorry, I'll leave a comment about in this in the next commit! Basically, I'm using Fuse.js extended search (https://fusejs.io/examples.html#extended-search) operator to search for items that include the pattern. Else, the default search uses fuzzy matching, so if I search something like "cube" there are a lot of extraneous search results.